spec/src/main/asciidoc/messaging-spec.adoc

```diff
@@ -5860,60 +5817,106 @@ Note that sendMessageNew does not need to throw JMSException.
 Here's how you might do this using the simplified API with the
 JMSContext created by injection:
 
+----
 @Inject
-
 @JMSConnectionFactory("jms/connectionFactory")
-
 private JMSContext context;
 
 @Resource(mappedName = "jms/dataQueue")
-
 private Queue dataQueue;
 
-public void sendMessageNew(String body) \{
-
-context.send(dataQueue, body);
-
+public void sendMessageNew(String body) {
+    context.send(dataQueue, body);
 }
+----
 
 === Sending a message (Java SE)
 
-This example compares the use of the classic and simplified JMS APIs for
-sending a TextMessage in a Java SE environment.
+This example compares the use of the classic and simplified Jakarta Messaging APIs for sending a TextMessage in a Java SE environment.
 
 ==== Example using the classic API
 
 Here’s how you might do this using the classic API:
 
-*public* *void* sendMessageOld(String body)
+public void sendMessageOld(String body)
+    throws JMSException, NamingException{
+
+    InitialContext namingContext = getInitialContext();
+    ConnectionFactory connectionFactory = (ConnectionFactory)
+    namingContext.lookup("jms/connectionFactory");
+    Queue dataQueue = (Queue) namingContext.lookup("jms/dataQueue")
+    try (Connection connection =
+            connectionFactory.createConnection()) {
+        Session session = connection.createSession();
+        MessageProducer messageProducer =
+        session.createProducer(dataQueue);
+        TextMessage textMessage = session.createTextMessage(body);
+        messageProducer.send(textMessage);
+    }
+}
 
-*throws* JMSException, NamingException\{
+In the above example, `getInitialContext()` is an application method which
+returns a suitable JNDI InitialContext.
 
-InitialContext namingContext = getInitialContext();
+==== Example using the simplified API
 
-ConnectionFactory connectionFactory = (ConnectionFactory)
+Here’s how you might do this using the simplified API:
 
-namingContext.lookup("jms/connectionFactory");
+----
+public void sendMessageNew(String body) throws NamingException {
 
-Queue dataQueue = (Queue) namingContext.lookup("jms/dataQueue")
+    InitialContext namingContext = getInitialContext();
+    ConnectionFactory connectionFactory = (ConnectionFactory)
+        namingContext.lookup("jms/connectionFactory");
+    Queue dataQueue = (Queue)namingContext.lookup("jms/dataQueue");
+    try (JMSContext context=connectionFactory.createContext();){
+        context.createProducer().send(dataQueue,body);
+    }
+}
 
```

Comment:
Missing closing block (----), this is messing rest of the document.

Reply:
I was afraid this could happen, I should've doublechecked. This is fixed and I also fixed it in one more place which Asciidoctor reported as missing ending element.